c++/src/capnp/layout.c++

```diff
@@ -1609,31 +1636,61 @@ struct WireHelpers {
       return { segment, ptr };
     } else {
       // List of structs.
+      WordCount declDataSize = roundBitsUpToWords(value.structDataSize);
```

Comment:
While I agree that no message encoded by the library as it stands should generate an inline composite for a non-word-multiple struct, afaict the `structDataSize` just comes off the tag word, and so can be whatever it wants. If someone makes an inline-composite of a byte struct, it might not be spec valid, but I'm not actually sure it would be prevented from making it to this point in the code.

In summary, I'm worried that a crafted message could make this assert trip.

Given this, can I use a `KJ_ASSERT()` instead of a `KJ_DASSERT()`? Otherwise I can just write the other-size tolerant code like I did for `setStructPointer`.